packages/fuel-indexer-lib/src/config/mod.rs

```diff
@@ -34,6 +34,8 @@ pub enum IndexerConfigError {
     ConfigFileError(#[from] Error),
     #[error("Error processing YAML file: {0:?}")]
     SerdeYamlError(#[from] serde_yaml::Error),
+    #[error("Error parsing database connection string into DatabaseConfig")]
```

Comment:
If we use `url` to parse the connection URI we might be able to derive `url::Error` here